Documentation/Testing/ProjectTesting.rst

```diff
@@ -6,262 +6,234 @@
-<https://www.drud.com/what-is-ddev/>`_ and add basic acceptance testing to it, executed
-locally and by GitHub Actions.
+**Projects** usually needs to support only one PHP version, Database vendor
+and version and TYPO3 core version
 
-This is thought as an inspiration you may want to adapt for your project.
+Version raises for upgrades are usually prepared on a branch and changed
+instead of parallel execution.
 
+Project may have different places for tests
 
-Project site-introduction
+*   local path extension tests `packages/*/Tests/*`
+*   global (root) tests `Tests/*`
+
+The Core mono repository is basically a project setup, having local path
+extensions in `typo3/sysexts/*` instead of the more known and lived `packages/*`
+project folder structure.
+
+.. _testing-projects-structure:
+
+Project structure
+=================
+
+We assume a project structure similar to `tf-basics-project <https://github.com/sbuerk/tf-basics-project>`__
+here. If you are using another structure, you have to adjust some scripts.
+
+*   :ref:`TYPO3 installation with Composer <t3start:install>` and default paths
+*   :ref:`DDEV <t3start:installation-ddev-tutorial>` is used for local development
+*   Local extensions and the site package are situated in :path:`packages`
+
+..  directory-tree::
+    :caption: Initial directory tree in version control
+    :level: 1
+
+    *   :path:`.ddev`
+    *   :path:`config`
+    *   :path:`packages`
+    *   :file:`composer.json`
+    *   :file:`composer.lock`
+
+The :file:`composer.json` looks like this:
+
+..  literalinclude:: _ProjectTesting/_composer.json
+    :caption: Example project composer.json before testing
+
+.. _testing-projects-installation:
+
+Install testing dependencies
+============================
+
+As a bare minimum it is suggested to use
+
+*   One coding style fixer for PHP, for example :composer:`friendsofphp/php-cs-fixer`
+*   One static code analyzer for PHP, for example :composer:`phpstan/phpstan`
+
+Depending on the complexity of your project you might need:
+
+*   :composer:`phpunit/phpunit`, if there is any PHP code of a complexity that
+    should be tested.
+*   Testing of scss, TypeScript or JavaScript (not covered here)
+*   Linting of YAML, XML, TypoScript (not covered here)
+*   :ref:`Writing acceptance tests <t3coreapi:testing-writing-acceptance>`
+
+You can install all these tools as development dependencies. They will then not
+be installed on your production system when Composer is executed with option
+`--no-dev` during deployment:
+
+..  code-block:: bash
+    :caption: Composer installation **during deployment**
+
+    composer install --no-dev
+
+For TYPO3 project you can use the package :composer:`typo3/coding-standards`
+which bundles :composer:`friendsofphp/php-cs-fixer`, :composer:`phpstan/phpstan`,
+:composer:`phpunit/phpunit` and some other helpful tools and comes with useful
+configurations for these.
```

Comment:
@sbuerk I remember a talk with you, that you do not install that package. Correct me, if I quote you wrong: "That package would be nice, if there would be someone keeping the work up-2-date. Currenty, you can just install php-cs-fixer as dependency, take the config files from core and you're done. So, no need to install that package"

Reply:
Yes you can but then You use the advantage of a common package

Reply:
I would also rather like to encourage people to not use a bundled install, but maintain the configs here, so that they understand each tool and are able to react if a "meta package" isn't used.

We can hint at that package for inspiration or an alternate approach but maybe not encourage to use that one?

Reply:
> @sbuerk I remember a talk with you, that you do not install that package. Correct me, if I quote you wrong: "That package would be nice, if there would be someone keeping the work up-2-date. Currenty, you can just install php-cs-fixer as dependency, take the config files from core and you're done. So, no need to install that package"

That as a purly personal preference and point of view, without the intention to blame anybody. It's till a legit packagit to be used if it suits your/someones need. My personal view here should not be taken as a general recommandation - and I thought I had made that clear.

Reply:
Maybe having a dedicated section somewhere related to `code style linting/fixing` and providing/explaining usual used tools - raw or with the `typo3/coding-standards` package. Developers/Teams can than decide on their preferences what to use and having the guide at hand - and link from here.

Reply:
Such a section is a good idea, but as long as we don't have one using the one from `typo3/coding-standards` is certainly better then none. And as the topic is opinionated anyway I would stick with. Use whatever you want, for example you can use..

Reply:
Once the section exists we can link it from here

Reply:
I added a todo for that.